operator/controllers/operator/eventexporter_controller.go

```diff
@@ -0,0 +1,210 @@
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+
+func (r *EventExporterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := runtimelog.FromContext(ctx)
+	log.Info(fmt.Sprintf("===============eventexporter reconcile started (ns: %s, name: %s)===============", req.Namespace, req.Name))
+
+	eventExporter := operatorv1alpha1.EventExporter{}
+	if err := r.Client.Get(ctx, req.NamespacedName, &eventExporter); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if _, err := r.overlayData(ctx, log, &eventExporter); err != nil {
+		l.Error(err, "failed to overlay eventexporter's configMap")
+		return ctrl.Result{}, err
+	}
+
+	ff, err := r.FileRepo.GetFilesRecursive("templates")
+	if err != nil {
+		log.Error(err, "failed to load resource templates")
+		return ctrl.Result{}, err
+	}
+
+	app := kubernetes.Application{
+		Client:   r.Client,
+		FileRepo: r.FileRepo,
+		CR:       &eventExporter,
+		GVK:      operatorv1alpha1.GroupVersion.WithKind("EventExporter"),
+		Recorder: r.Recorder,
+		TmplFunc: template.FuncMap{
+			"md5Data": func() string { return MD5Hash(eventExporter.Spec.Config) },
+		},
+	}
+
+	if err := app.ApplyAll(ctx, ff, log); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.checkState(ctx, log, &eventExporter); err != nil {
+		l.Error(err, "failed to check sub resources state")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{RequeueAfter: schedDuration}, nil
+
+}
+
+func (r *EventExporterReconciler) overlayData(ctx context.Context, log logr.Logger, eventExporter *operatorv1alpha1.EventExporter) (changed bool, err error) {
+
+	configmap := core.ConfigMap{}
+	err = r.Client.Get(ctx, client.ObjectKey{Namespace: eventExporter.Namespace, Name: eventExporter.Name}, &configmap)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, "failed to get the eventexporter's configmap")
+		return false, err
+	}
+
+	newMd5 := MD5Hash(eventExporter.Spec.Config)
+	oldMd5 := MD5Hash("")
+	if !apierrors.IsNotFound(err) {
+		oldMd5 = configmap.Labels["md5-data"]
+	}
+
+	if newMd5 == oldMd5 {
+		log.Info("eventexporter configuration keeps the same as before")
+		return false, nil
+	}
+
+	if !apierrors.IsNotFound(err) {
+		if err = r.Client.Delete(ctx, &configmap); err != nil {
+			log.Error(err, "failed to delete eventexporter's configmap")
+			return true, err
+		}
+	}
+
+	configmap = core.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      eventExporter.Name,
+			Namespace: eventExporter.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: eventExporter.APIVersion,
+					Kind:       eventExporter.Kind,
+					Name:       eventExporter.Name,
+					UID:        eventExporter.UID,
+				},
+			},
+			Labels: map[string]string{
+				"version":  eventExporter.Spec.Version,
+				"md5-data": newMd5,
+			},
+		},
+		Data: map[string]string{"config.yaml": eventExporter.Spec.Config},
```

Comment:
If we create a new configmap for every new config in the spec, we can make the configmap immutable: https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-immutable

Reply:
Thx, updated!